ui/src/notebooks/pipes/markdown/MarkdownPanel.tsx

```diff
@@ -0,0 +1,48 @@
+// Libraries
+import React, {FC} from 'react'
+
+// Types
+import {PipeProp} from 'src/notebooks'
+import {MarkdownMode} from './'
+
+// Components
+import MarkdownModeToggle from './MarkdownModeToggle'
+import MarkdownPanelEditor from './MarkdownPanelEditor'
+import {MarkdownRenderer} from 'src/shared/components/views/MarkdownRenderer'
+
+const MarkdownPanel: FC<PipeProp> = ({data, Context, onUpdate}) => {
+  const handleToggleMode = (mode: MarkdownMode): void => {
+    const updatedData = {...data, mode}
+    
+    onUpdate(updatedData)
+  }
+
+  const controls = (
+    <MarkdownModeToggle mode={data.mode} onToggleMode={handleToggleMode} />
+  )
+
+  const handleChange = (text: string): void => {
+    const updatedData = {...data, text}
+
+    onUpdate(updatedData)
+  }
+
+  let panelContents = (
+    <MarkdownPanelEditor text={data.text} onChange={handleChange} />
+  )
+
+  if (data.mode === 'preview') {
```

Comment:
my inner watts is saying that the wrapped renderer and the editor are two different components with a `data.mode` guard in each of them so that you'd add both as children, and they choose to toggle